src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -42,9 +43,11 @@
   private static final String SETTINGS_TEMPLATE_FILENAME = "java/settings.snip";
   private static final String STUB_SETTINGS_TEMPLATE_FILENAME = "java/stub_settings.snip";
   private static final String STUB_INTERFACE_TEMPLATE_FILENAME = "java/stub_interface.snip";
-  private static final String GRPC_STUB_TEMPLATE_FILENAME = "java/grpc_stub.snip";
+  private static final String STUB_TEMPLATE_FILENAME = "java/stub.snip";
```

Comment:
Since only one of HTTP or gRPC will be generated, would it make sense to split the template into `grpc_stub.snip` and `http_stub.snip`? As a reader, I would prefer duplicate code to parsing the diffs (which is less clear without doing a deep dive).

Ditto for `test.snip`.

Reply:
GRPC and REST stubs share too much (like 90-80% of the file is the same). Keeping them separate makes it much easier for rest and grpc stubs to drift from each other leading to troubles supporting it. 

I tried to make the .ship files match the philosophical view on the two different transports implementations - they are essentially the same thing with some transport-specific tweaks. Duplicating the .snip files for each transport views them differently - two transports are two separate independent things.

Reply:
SGTM.

Reply:
+1